### PR TITLE
Add tests for Symbol.prototype.description

### PR DESCRIPTION
**Pull request by joyeecheung**
Refs: https://github.com/tc39/proposal-Symbol-description/issues/5
V8 implementation in https://chromium-review.googlesource.com/c/v8/v8/+/1088871

### CODE REVIEW

**Comment by joyeecheung** [comment]
cc @michaelficarra @gsathya @littledan

File: test/built-ins/Symbol/prototype/description/descriptor.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+info: |
+    `Symbol.prototype.description` is an accessor property whose
+    set accessor function is undefined.
+features: [Symbol.prototype.description]
+---*/
+
+const desc = Object.getOwnPropertyDescriptor(Symbol.prototype, 'description');
+assert.sameValue(typeof desc.get, 'function');
+assert.sameValue(desc.set, undefined);
+assert.sameValue(desc.writable, undefined);
```
Comment (ljharb) [inline test/built-ins/Symbol/prototype/description/descriptor.js]: There’s a propertyHelper that could be used here

File: test/built-ins/Symbol/prototype/description/get.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    2. Let sym be ? thisSymbolValue(s).
+    3. Return sym.[[Description]].
+features: [Symbol.prototype.description]
+---*/
+
+const symbol = Symbol('test');
+assert.sameValue(symbol.description, 'test');
+assert.sameValue(symbol.hasOwnProperty('description'), false);
+
+const empty = Symbol();
```
Comment (ljharb) [inline test/built-ins/Symbol/prototype/description/get.js]: Could you also test `Symbol(undefined)` and `Symbol(‘’)`?

File: test/built-ins/Symbol/prototype/description/this-val-non-symbol.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    slot.
+info: |
+    1. Let s be the this value.
+    2. Let sym be ? thisSymbolValue(s).
+    3. Return sym.[[Description]].
+features: [Symbol.prototype.description]
+---*/
+
+assert.throws(TypeError, function() {
+  Symbol.prototype.description.call(null);
```
Comment (michaelficarra) [inline test/built-ins/Symbol/prototype/description/this-val-non-symbol.js]: This fails, but for the wrong reason. The getter is being called on `Symbol.prototype`. You should instead extract the getter with `Object.getOwnPropertyDescriptor`.

**Comment by joyeecheung** [comment]
@ljharb @michaelficarra Thanks for the reviews. I have updated the PR, PTAL.

**Comment by joyeecheung** [comment]
> LGTM - we may also want to add some tests that .call-ing the getter on real symbols works too?

@ljharb Sure! I've updated the PR, PTAL.

**Pull request merged**